### PR TITLE
🌐 [i18n-KO] Translated `training.mdx` to Korean

### PR DESCRIPTION
**Pull request by gabrielwithappy**
# What does this PR do?

Translated the `training.mdx` file of the documentation to Korean.
Thank you in advance for your review. 

Part of https://github.com/huggingface/transformers/issues/20179

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
Team PseudoLab, may you please review this PR?
@0525hhgus, @KIHOON71, @sim-so, @gabrielwithappy, @HanNayeoniee, @wonhyeongseo, @jungnerd

 May you please review this PR?
@sgugger, @ArthurZucker, @eunseojo

### CODE REVIEW

File: docs/source/ko/training.mdx [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+```
+</pt>
+<tf>
+<a id='keras'></a>
+
+<Youtube id="rnTGBy2ax1c"/>
+
+## Keras로 텐서플로우 모델 훈련하기[[Train a TensorFlow model with Keras]]
+
+Keras API를 사용하여 텐서플로우우에서 🤗 트랜스포머 모델을 훈련할 수도 있습니다!
```
Comment (HanNayeoniee) [inline docs/source/ko/training.mdx:163]: Keras API를 사용하여 텐서플로우에서 🤗 트랜스포머 모델을 훈련할 수도 있습니다!

'우우'가 두 번 들어갔네요

**Review by HanNayeoniee — CHANGES_REQUESTED** [review]

File: docs/source/ko/training.mdx (hunk 1 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:163]: 오옷 찾아주셔소 감감사합니다. 🙂🙏

**Comment by gabrielwithappy** [comment]
Team PseudoLab, may you please review this PR?
@0525hhgus, @KIHOON71, @sim-so, @gabrielwithappy, @HanNayeoniee, @wonhyeongseo, @jungnerd

File: docs/source/ko/autoclass_tutorial.mdx [hunk 2]
```diff
@@ -12,7 +12,8 @@ specific language governing permissions and limitations under the License.
 
 # AutoClass로 사전 학습된 인스턴스 로드[[Load pretrained instances with an AutoClass]]
 
-트랜스포머 아키텍처가 매우 다양하기 때문에 체크포인트에 맞는 아키텍처를 생성하는 것이 어려울 수 있습니다. 라이브러리를 쉽고 간단하며 유연하게 사용하기 위한 Transformer 핵심 철학의 일환으로, `AutoClass`는 주어진 체크포인트에서 올바른 아키텍처를 자동으로 추론하여 로드합니다. `from_pretrained()` 메서드를 사용하면 모든 아키텍처에 대해 사전 학습된 모델을 빠르게 로드할 수 있으므로 모델을 처음부터 학습하는 데 시간과 리소스를 투입할 필요가 없습니다. 이러한 유형의 체크포인트에 구애받지 않는 코드를 생성한다는 것은 코드가 한 체크포인트에서 작동한다면 아키텍처가 다르더라도 유사한 작업에 대해 학습된 것이라면 다른 체크포인트에서도 작동한다는 것을 의미합니다.
+트랜스포머 아키텍처가 매우 다양하기 때문에 체크포인트에 맞는 아키텍처를 생성하는 것이 어려울 수 있습니다. 라이브러리를 쉽고 간단하며 유연하게 사용하기 위한 Transformer 핵심 철학의 일환으로, `AutoClass`는 주어진 체크포인트에서 올바른 아키텍처를 자동으로 추론하여 로드합니다. `from_pretrained()` 메서드를 사용하면 모든 아키텍처에 대해 사전 학습된 모델을 빠르게 로드할 수 있으므로 모델을 처음부터 학습하는 데 시간과 리소스를 투입할 필요가 없습니다. 
+체크포인트에 구애받지 않는 코드를 생성한다는 것은 코드가 한 체크포인트에서 작동하면 아키텍처가 다르더라도 다른 체크포인트(유사한 작업에 대해 학습된 경우)에서도 작동한다는 것을 의미합니다.
```
Comment (wonhyeongseo) [inline docs/source/ko/autoclass_tutorial.mdx:16]: Love the rewrite, sir! 🥰

File: docs/source/ko/training.mdx [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+* Keras를 사용하여 TensorFlow에서 사전 학습된 모델을 미세 튜닝하기.
+* 기본 PyTorch에서 사전 학습된 모델을 미세 튜닝하기.
+
+<a id='data-processing'></a>
+
+## 데이터셋 준비[[Prepare a dataset]]
+
+<Youtube id="_BZearw7f0w"/>
+
+사전 학습된 모델을 미세 튜닝하기 위해서 데이터셋을 다운로드하고 훈련할 수 있도록 준비합니다. 이전 튜토리얼에서 훈련련을 위해 데이터를 처리하는 방법을 보여드렸는데, 이제 그 기술을 시험해볼 수 있는 기회가 생겼습니다!
```
Comment (wonhyeongseo) [inline docs/source/ko/training.mdx:29]:
```suggestion
사전 학습된 모델을 미세 조정하기 위해서 데이터셋을 다운로드하고 훈련할 수 있도록 준비하세요. 이전 튜토리얼에서 훈련을 위해 데이터를 처리하는 방법을 보여드렸는데, 지금이 배운걸 되짚을 기회입니다!
```
"훈련련"을 교정하려다 조금 더 수정해보았어요! 의역이다 보니 참고만 해주시길 바랍니당 🙏

File: docs/source/ko/training.mdx [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+```
+
+텍스트를 처리하고 가변 시퀀스 길이를 처리하기 위한 패딩 및 잘라내기 전략을 포함하려면 토큰화 도구가 필요합니다. 데이터셋을 한 번에 처리하려면 🤗 데이터셋 [`map`](https://huggingface.co/docs/datasets/process.html#map) 메서드를 사용하여 전체 데이터셋에 전처리 함수를 적용합니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/training.mdx:42]:
```suggestion
텍스트를 처리하고 시퀀스의 서로 다른 길이에 따른 패딩 및 잘라내기 전략을 포함하려면 토크나이저가 필요합니다. 데이터셋을 한 번에 처리하려면 🤗 Datasets의 [`map`](https://huggingface.co/docs/datasets/process.html#map) 메서드를 사용하여 전체 데이터셋에 전처리 함수를 적용하세요:
```
🤗 뒤에 나오는 단어는 제품명입니당

File: docs/source/ko/training.mdx [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+
+여기서 부터는 사용하려는 프레임워크에 해당하는 섹션을 따라야 합니다. 오른쪽 사이드바의 링크를 사용하여 원하는 프레임워크로 이동할 수 있으며, 특정 프레임워크의 모든 콘텐츠를 숨기려면 해당 프레임워크 블록의 오른쪽 상단에 있는 버튼을 사용하면 됩니다!
+
+<frameworkcontent>
+<pt>
+<Youtube id="nvBXf7s7vTI"/>
+
+## 파이토치 Trainer로 훈련하기[[Train with PyTorch Trainer]]
+
+🤗 트랜스포머는 🤗 트랜스포머 모델 훈련에 최적화된 [`Trainer`] 클래스를 제공하여 훈련 루프를 직접 작성하지 않고도 쉽게 훈련을 시작할 수 있습니다. 🤗 트랜스포머는 🤗 트랜스포머 모델 훈련에 최적화된 [`Trainer`] 클래스를 제공하여 훈련 루프를 직접 작성하지 않고도 쉽게 훈련을 시작할 수 있습니다. [`Trainer`] API는 logging, 그래디언트 누적(gradient accumulation), 혼합 정밀도(mixed precision) 등 다양한 훈련 옵션과 기능을 지원합니다.
```
Comment (wonhyeongseo) [inline docs/source/ko/training.mdx:76]:
```suggestion
🤗 Transformers는 🤗 Transformers 모델 훈련에 최적화된 [`Trainer`] 클래스를 제공하여 훈련 루프를 직접 작성하지 않고도 쉽게 훈련을 시작할 수 있습니다. [`Trainer`] API는 로깅(logging), 경사 누적(gradient accumulation), 혼합 정밀도(mixed precision) 등 다양한 훈련 옵션과 기능을 지원합니다.
```
중복 제거 및 제품명을 사용했어요!
경사 누적에 관련된 문서:
- [🤗](https://huggingface.co/docs/accelerate/usage_guides/gradient_accumulation)
- [towards data) science](https://towardsdatascience.com/gradient-accumulation-overcoming-memory-constraints-in-deep-learning-36d411252d01)

File: docs/source/ko/training.mdx [hunk 6]
```diff
@@ -0,0 +1,425 @@
...
+
+먼저 모델을 로드하고 예상되는 레이블 수를 지정합니다. Yelp 리뷰 [데이터셋 카드](https://huggingface.co/datasets/yelp_review_full#data-fields)에서 5개의 레이블이 있음을 알 수 있습니다:
```
Comment (wonhyeongseo) [inline docs/source/ko/training.mdx:78]:
```suggestion
먼저 모델을 가져오고 예상되는 레이블 수를 지정합니다. Yelp 리뷰 [데이터셋 카드](https://huggingface.co/datasets/yelp_review_full#data-fields)에서 5개의 레이블이 있음을 알 수 있습니다:
```

File: docs/source/ko/training.mdx [hunk 7]
```diff
@@ -0,0 +1,425 @@
...
+```py
+>>> from transformers import AutoModelForSequenceClassification
+
+>>> model = AutoModelForSequenceClassification.from_pretrained("bert-base-cased", num_labels=5)
+```
+
+<Tip>
+
+사전 훈련된 가중치 중 일부가 사용되지 않고 일부 가중치가 무작위로 표시된다는 경고가 표시됩니다.
+걱정하지 마세요. 이것은 완전히 정상입니다! 사전 학습된 BERT 모델의 헤드는 폐기되고 무작위로 초기화된 분류 헤드로 대체됩니다. 이제 사전 학습된 모델의 지식을 이용하여 시퀀스 분류 작업을 위한 새로운 모델 헤드를 미세 튜닝 합니다.
```
Comment (wonhyeongseo) [inline docs/source/ko/training.mdx:89]:
```suggestion
걱정마세요. 원래 그렇습니다! 사전 학습된 BERT 모델의 헤드는 폐기되고 무작위로 초기화된 분류 헤드로 대체됩니다. 이제 시퀀스 분류 작업을 위한 새로운 모델 헤드를 사전 학습된 모델의 지식으로 미세 조정해야 합니다.
```

**Review by wonhyeongseo — CHANGES_REQUESTED** [review]
~L100 preliminary review
Thank you so much for your hard work, @gabrielwithappy 💜

File: docs/source/ko/training.mdx (hunk 3 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:29]: 훈련은 수정할께요. 미세 튜닝은 NLP책에서 사용해서 미세 튜닝으로 사용하고 있습니다 🙂

File: docs/source/ko/training.mdx (hunk 7 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:89]: 이제 시퀀스 분류 작업을 위한 새로운 모델 헤드를 사전 학습된 모델의 지식으로 미세 조정해야 합니다.
마지막 문장에는 `모델 헤드`가 들어가야 될 것 같습니다. 이전 문장에서 BERT모델의 헤드가 폐기된 후 새롭게 미세 튜닝되는 대상이 `모델 헤드`인 것 같아요.

File: docs/source/ko/training.mdx (hunk 5 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx]: 수정했습니다 :-)

File: docs/source/ko/training.mdx (hunk 6 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx]: 수정완료 :-)

File: docs/source/ko/training.mdx [hunk 8]
```diff
@@ -0,0 +1,425 @@
...
+# Lower learning rates are often better for fine-tuning transformers
+model.compile(optimizer=Adam(3e-5))
+
+model.fit(tokenized_data, labels)
+```
+
+<Tip>
+
+모델을 `compile()`할 때 손실 인수를 모델에 전달할 필요가 없습니다! 
+이 인자를 비워두면 허깅 페이스 모델은 작업과 모델 아키텍처에 적합한 손실을 자동으로 선택합니다.
```
Comment (sim-so) [inline docs/source/ko/training.mdx:211]: Both "인수" and "인자" is used for "argument" in this document. I think we decided on "인수" for "argument," but let us discuss more if you want. 😄

File: docs/source/ko/training.mdx [hunk 9]
```diff
@@ -0,0 +1,425 @@
...
+
+<Tip>
+
+모델을 `compile()`할 때 손실 인수를 모델에 전달할 필요가 없습니다! 
+이 인자를 비워두면 허깅 페이스 모델은 작업과 모델 아키텍처에 적합한 손실을 자동으로 선택합니다. 
+원한다면 언제든지 직접 손실을 지정하여 이를 재정의할 수 있습니다!
+
+</Tip>
+
+이 접근 방식은 소규모 데이터 집합에서는 잘 작동하지만, 대규모 데이터 집합에서는 문제가 되기 시작할 수 있습니다. 왜 그럴까요?
```
Comment (sim-so) [inline docs/source/ko/training.mdx:216]:
```suggestion
이 접근 방식은 소규모 데이터 집합에서는 잘 작동하지만, 대규모 데이터 집합에서는 문제가 될 수 있습니다. 왜 그럴까요?
```
The question mark was also fixed!

File: docs/source/ko/training.mdx [hunk 10]
```diff
@@ -0,0 +1,425 @@
...
+토큰화된 배열과 레이블을 메모리에 완전히 로드해야 하고, NumPy는 "들쭉날쭉한" 배열을 처리하지 않기 때문입니다.
+"들쭉날쭉한" 배열을 처리하지 않기 때문에, 모든 토큰화된 샘플은 전체 데이터셋에서 가장 긴 샘플의 길이만큼 추가되어야 합니다.
+데이터셋의 길이로 채워야 합니다. 이렇게 하면 배열이 훨씬 더 커지고 이런 패딩 토큰으로 인해 학습 속도도 느려집니다!
```
Comment (sim-so) [inline docs/source/ko/training.mdx:219]: I love "들쭉날쭉" 😆 ❤️ 
```suggestion
토큰화된 배열과 레이블을 메모리에 완전히 로드해야 하고, NumPy는 "들쭉날쭉한" 배열을 처리하지 않기 때문에, 
모든 토큰화된 샘플을 전체 데이터셋에서 가장 긴 샘플의 길이만큼 패딩해야 합니다. 
이렇게 하면 배열이 훨씬 더 커지고, 이 패딩 토큰으로 인해 학습 속도도 느려집니다!
```
I would suggest you remove some duplicated words.

File: docs/source/ko/training.mdx [hunk 11]
```diff
@@ -0,0 +1,425 @@
...
+
+### 데이터를 tf.data.Dataset으로 로드하기[[Loading data as a tf.data.Dataset]]
+
+학습 속도가 느려지는 것을 피하려면 대신 데이터를 `tf.data.Dataset`으로 로드할 수 있습니다. 원한다면 직접
+tf.data` 파이프라인을 직접 작성할 수도 있지만, 이 작업을 수행하는 데 편리한 두 가지 방법이 있습니다:
```
Comment (sim-so) [inline docs/source/ko/training.mdx:224]:
```suggestion
학습 속도가 느려지는 것을 피하려면 데이터를 `tf.data.Dataset`로 로드할 수 있습니다. 
원한다면 직접 `tf.data` 파이프라인을 작성할 수도 있지만, 이 작업을 간편하게 수행할 수 있는 두 가지 방법이 있습니다:
```

File: docs/source/ko/training.mdx [hunk 12]
```diff
@@ -0,0 +1,425 @@
...
+
+- [`~TFPreTrainedModel.prepare_tf_dataset`]: 대부분의 경우 이 방법을 권장합니다. 모델의 메서드이기 때문에 모델을 검사하여 모델 입력으로 사용할 수 있는 열을 자동으로 파악하고
+나머지는 버려서 더 단순하고 성능이 좋은 데이터 집합을 만들 수 있습니다.
+- [`~datasets.Dataset.to_tf_dataset`]: 이 방법은 좀 더 낮은 수준이며, 포함할 '열'과 '레이블'을 정확히 지정하여
+데이터셋을 생성하는 방법을 정확히 제어하고 싶을 때 유용하며, 포함할 'columns'과 'label_cols'을 정확히 지정할 수 있습니다.
+
+[`~TFPreTrainedModel.prepare_tf_dataset`]을 사용하려면 먼저 다음 코드 샘플과 같이 토큰화기 출력을 데이터 세트에 열로 추가해야 합니다:
```
Comment (sim-so) [inline docs/source/ko/training.mdx:231]:
```suggestion
[`~TFPreTrainedModel.prepare_tf_dataset`]을 사용하려면 먼저 다음 코드 샘플과 같이 토크나이저 출력을 데이터 세트에 열로 추가해야 합니다:
```

File: docs/source/ko/training.mdx [hunk 13]
```diff
@@ -0,0 +1,425 @@
...
+def tokenize_dataset(data):
+    # Keys of the returned dictionary will be added to the dataset as columns
+    return tokenizer(data["text"])
+
+
+dataset = dataset.map(tokenize_dataset)
+```
+
+허깅 페이스 데이터 세트는 기본적으로 디스크에 저장되므로 메모리 사용량을 늘리지 않는다는 점을 기억하세요! 
+열이 추가되면 데이터 세트에서 배치를 스트리밍하고 각 배치에 패딩을 추가할 수 있으므로 전체 데이터셋셋에 패딩을 추가하는 것보다 패딩 토큰의 수를 크게 줄일 수 있습니다.
```
Comment (sim-so) [inline docs/source/ko/training.mdx:243]:
```suggestion
허깅 페이스 데이터셋은 기본적으로 디스크에 저장되므로 메모리 사용량을 늘리지 않는다는 점을 기억하세요! 
열이 추가되면 데이터셋에서 배치를 스트리밍하고 각 배치에 패딩을 추가할 수 있으므로 전체 데이터셋에 패딩을 추가하는 것보다 패딩 토큰의 수를 크게 줄일 수 있습니다.
```

File: docs/source/ko/training.mdx [hunk 14]
```diff
@@ -0,0 +1,425 @@
...
+
+
+```py
+>>> tf_dataset = model.prepare_tf_dataset(dataset, batch_size=16, shuffle=True, tokenizer=tokenizer)
+```
+
+위의 코드 샘플에서는 배치가 로드될 때 올바르게 패딩할 수 있도록 `prepare_tf_dataset`에 토큰나이저를 전달해야 합니다.
```
Comment (sim-so) [inline docs/source/ko/training.mdx:250]:
```suggestion
위의 코드 샘플에서는 배치가 로드될 때 올바르게 패딩할 수 있도록 `prepare_tf_dataset`에 토크나이저를 전달해야 합니다.
```

File: docs/source/ko/training.mdx [hunk 15]
```diff
@@ -0,0 +1,425 @@
...
+데이터 세트의 모든 샘플 길이가 같고 패딩이 필요하지 않은 경우 이 인수를 건너뛸 수 있습니다.
```
Comment (sim-so) [inline docs/source/ko/training.mdx:251]:
```suggestion
데이터셋의 모든 샘플 길이가 같고 패딩이 필요하지 않은 경우 이 인수를 건너뛸 수 있습니다.
```

File: docs/source/ko/training.mdx [hunk 16]
```diff
@@ -0,0 +1,425 @@
...
+샘플을 채우는 것보다 더 복잡한 작업을 수행해야 하는 경우(예: 마스킹된 언어의 토큰 손상 모델링)을 위해 토큰을 손상시켜야 하는 경우, 
+대신 `collate_fn` 인자를 사용하여 호출될 함수를 전달할 수 있습니다. 샘플 목록을 일괄 처리로 변환하고 원하는 전처리를 적용할 함수를 전달할 수 있습니다.
```
Comment (sim-so) [inline docs/source/ko/training.mdx:253]:
```suggestion
샘플을 채우는 것보다 더 복잡한 작업(예: 마스킹된 언어의 토큰 손상 모델링)을 수행하기 위해 토큰을 손상시켜야 하는 경우, 
대신 `collate_fn` 인수를 사용하여 샘플 목록을 배치로 변환하고 원하는 전처리를 적용할 함수를 전달할 수 있습니다. 
```

**Review by sim-so — COMMENTED** [review]
I reviewed ~ line 268.
I will review the rest soon!
Thank you for your hard work!

**Review by sim-so — CHANGES_REQUESTED** [review]
I reviewed ~ line 268.
I will review the rest soon!
Thank you for your hard work!

File: docs/source/ko/training.mdx [hunk 17]
```diff
@@ -0,0 +1,425 @@
...
+```py
+>>> from transformers import TrainingArguments
+
+>>> training_args = TrainingArguments(output_dir="test_trainer")
+```
+
+### Evaluate
+
+[`Trainer`]는 훈련 중에 모델 성능을 자동으로 평가하지 않습니다. 평가 지표를 계산하고 보고할 함수를 [`Trainer`]에 전달해야 합니다. 
+[🤗 Evaluate](https://huggingface.co/docs/evaluate/index) 라이브러리는 [`evaluate.load`](https://huggingface.co/spaces/evaluate-metric/accuracy) 함수로 로드할 수 있는 간단한 [`accuracy`]함수를 제공합니다 (자세한 내용은 [quicktour](https://huggingface.co/docs/evaluate/a_quick_tour) 함수를 참조하세요):
```
[the remark this answers is not in the document]
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:110]: 수정했습니다 .리뷰 감사드립니다. :-)

File: docs/source/ko/training.mdx (hunk 13 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:243]: fixed, Thank you :-)

File: docs/source/ko/training.mdx (hunk 15 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:251]: fixed, Thank you :-)

File: docs/source/ko/training.mdx [hunk 18]
```diff
@@ -0,0 +1,425 @@
...
+간단한 바이너리 텍스트 분류 작업이므로 지금은 훈련 데이터 분할만 사용합니다.
+
+```py
+from datasets import load_dataset
+
+dataset = load_dataset("glue", "cola")
+dataset = dataset["train"]  # Just take the training split for now
+```
+
+다음으로 토크나이저를 로드하고 데이터를 NumPy 배열로 토큰화합니다. 레이블은 이미 0과 1의 목록이기 때문에 토큰화하지 않고 바로 NumPy 배열로 변환할 수 있습니다!
```
[the remark this answers is not in the document]
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:181]: fixed, thank you. :-)

File: docs/source/ko/training.mdx (hunk 8 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:211]: fixed, thank you.

File: docs/source/ko/training.mdx (hunk 9 above)
Comment (gabrielwithappy) [reply docs/source/ko/training.mdx:216]: fixed, thank you. :-)

**Comment by gabrielwithappy** [comment]
May you please review this PR?
@sgugger, @ArthurZucker, @eunseojo

**Review by sgugger — APPROVED** [review]
Thanks for your contribution!

**Pull request merged**